src/router/__tests__/legacyPermalinkManagement.routerPlugin.spec.js

```diff
@@ -0,0 +1,366 @@
+            handleLegacyParams({ param, legacyCoordinates })
+
+            expect(legacyCoordinates).to.deep.equal([0, legacyValue])
+        })
+        it('X', () => {
+            const param = 'X'
+            const legacyCoordinates = [0, 0]
+
+            handleLegacyParams({ param, legacyCoordinates })
+
+            expect(legacyCoordinates).to.deep.equal([0, legacyValue])
+        })
+        it('E', () => {
+            const param = 'E'
+            const legacyCoordinates = [0, 0]
+
+            handleLegacyParams({ param, legacyCoordinates })
+
+            expect(legacyCoordinates).to.deep.equal([legacyValue, 0])
+        })
+        it('Y', () => {
+            const param = 'Y'
+            const legacyCoordinates = [0, 0]
+
+            handleLegacyParams({ param, legacyCoordinates })
+
+            expect(legacyCoordinates).to.deep.equal([legacyValue, 0])
+        })
+        it('lon', () => {
+            const param = 'lon'
+            const legacyValue = '10'
+            const latlongCoordinates = [0, 0]
+
+            handleLegacyParams({ param, latlongCoordinates, legacyValue })
+
+            expect(latlongCoordinates).to.deep.equal([Number(legacyValue), 0])
+            expect(cameraPosition).to.deep.equal([
+                Number(legacyValue),
+                null,
+                null,
+                null,
+                null,
+                null,
+            ])
+        })
+        it('lat', () => {
+            const param = 'lat'
+            const legacyValue = '10'
+            const latlongCoordinates = [0, 0]
+
+            handleLegacyParams({ param, latlongCoordinates, legacyValue })
+
+            expect(latlongCoordinates).to.deep.equal([0, Number(legacyValue)])
+            expect(cameraPosition).to.deep.equal([
+                null,
+                Number(legacyValue),
+                null,
+                null,
+                null,
+                null,
+            ])
+        })
+        it('layers with new separators', () => {
+            const param = 'layers'
+            const legacyValue = ['@feature']
+            vi.spyOn(utils, 'getLayersFromLegacyUrlParams')
+
+            handleLegacyParams({ param, legacyValue })
+
+            expect(newQuery).to.deep.equal({ layers: ['@feature'] })
+            expect(utils.getLayersFromLegacyUrlParams).not.toHaveBeenCalled()
+        })
+        it('layers without new separators', () => {
+            const fakeStore = {
+                state: {
+                    layers: {
+                        config: [layerConfig],
+                    },
+                    position: {},
+                },
+                getters: {
+                    getLayerConfigById: (_) => {
+                        layerConfig
+                    },
+                },
+            }
+            const layerVisibility = 'layers_visibility'
+            const layerOpacity = 'layers_opacity'
+            const layerTimestamp = 'layers_timestamp'
```

Comment:
if we want to check it's transformed correctly, we should also make tests which checks that the values are transformed correctly. This means : 
we need to set the visibility to 'true'
we need to check that if we set the opacity to a valid value (a value between 0 and 1 as a string)
we need to add a timeConfig to the layerConfig and a timestamp to check that it can be added to the layer

but it is also good to check that if we set invalid values, it work as expected and ignores those values.

Reply:
I think this has to be done in a unit test of the transformLayerIntoUrlString function and does not belong to this test
do you want me to create also a unit test for the transformLayerIntoUrlString function ?

Reply:
Well, handleLegacyParameter uses transformLayerIntoUrlString anyway, but we should also test the transformLayerIntoUrlString if it's not already tested.